aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -135,15 +136,32 @@ Tensor angle(const Tensor& self) {
 }
 
 Tensor real(const Tensor& self) {
-  TORCH_CHECK(!self.is_complex(), "real is not yet implemented for complex tensors.");
-  return self;
+  if (self.is_complex()) {
+    auto size = self.sizes().vec();
+    if (self.numel() == 0) {
+      return at::empty({0}, self.options().dtype(c10::toValueType(self.scalar_type())));
+    } else {
+      auto float_tensor = at::native::view_complex_as_float(self);
+      return at::narrow(float_tensor, float_tensor.dim() - 1, 0, 1).view(self.sizes());
```

Comment:
You should use `at::select()` here, no need to have an extra argument and a dimension of size 1.